src/app/loadout/fashion/FashionDrawer.tsx

```diff
@@ -0,0 +1,478 @@
+    item,
+    socket,
+    plugHash,
+  }: {
+    item: DimItem;
+    socket: DimSocket;
+    plugHash: number;
+  }) => {
+    setModsByBucket((modsByBucket) => {
+      // Clear out existing selections for this socket.
+      const existingMods = (modsByBucket[item.bucket.hash] ?? []).filter(
+        (mod) =>
+          mod !== socket.socketDefinition.singleInitialItemHash &&
+          !socket.plugSet?.plugs.some((p) => p.plugDef.hash === mod)
+      );
+
+      return {
+        ...modsByBucket,
+        // Add in the new mod
+        [item.bucket.hash]: [...existingMods, plugHash],
+      };
+    });
+  };
+
+  const handleUseEquipped = () => {
+    const newModsByBucket = _.mapValues(armorItemsByBucketHash, (item) => {
+      const cosmeticSockets = item.sockets
+        ? getSocketsByCategoryHash(item.sockets, SocketCategoryHashes.ArmorCosmetics)
+        : [];
+      return _.compact(cosmeticSockets.map((s) => s.plugged?.plugDef.hash));
+    });
+
+    setModsByBucket(newModsByBucket);
+  };
+
+  const handleSyncShader = () => {
+    const shaders = Object.values(modsByBucket).flat().filter(isShader);
+
+    const groupedShaders = _.groupBy(shaders, (h) => h);
+    const mostCommonShaders = _.maxBy(Object.values(groupedShaders), (shaders) => shaders.length);
+    if (!mostCommonShaders) {
+      return;
+    }
+
+    setModsByBucket((modsByBucket) =>
+      Object.fromEntries(
+        LockableBucketHashes.map((bucketHash) => [
+          bucketHash,
+          [...(modsByBucket[bucketHash] ?? []).filter((h) => !isShader(h)), mostCommonShaders[0]],
+        ])
+      )
+    );
+  };
+
+  const handleSyncOrnament = () => {
+    const isShader = (h: number) =>
+      defs.InventoryItem.get(h)?.plug?.plugCategoryHash === PlugCategoryHashes.Shader;
+    const ornaments = Object.values(modsByBucket)
+      .flat()
+      .filter((h) => !isShader(h));
+    const groupedOrnaments = _.groupBy(ornaments, (h) => {
+      const collectibleHash =
+        defs.InventoryItem.get(h)?.collectibleHash ??
+        // if the item has no collectible hash, try to find an "identical" item with one
+        findOtherCopies(defs, h).find((i) => i.collectibleHash)?.collectibleHash;
+
+      return collectibleHash && defs.Collectible.get(collectibleHash)?.parentNodeHashes[0];
+    });
+    delete groupedOrnaments['undefined'];
+    const mostCommonOrnamentSet = _.maxBy(
+      Object.entries(groupedOrnaments),
+      ([_presentationHash, ornaments]) => ornaments.length
+    );
+    if (!mostCommonOrnamentSet) {
+      return;
+    }
+
+    const set = defs.PresentationNode.get(
+      parseInt(mostCommonOrnamentSet[0], 10)
+    ).children.collectibles.map(
+      (c) =>
+        defs.Collectible.get(c.collectibleHash).itemHash ??
+        manuallyFindItemForCollectible(defs, c.collectibleHash)
+    );
+
+    setModsByBucket((modsByBucket) =>
+      Object.fromEntries(
+        LockableBucketHashes.map((bucketHash, i) => {
+          let ornamentHash = set[i];
```

Comment:
there's a handful of collections nodes i think this index based approach will error on
![image](https://user-images.githubusercontent.com/68782081/147991391-fdddfbd1-6894-4de9-b66b-ffba3df36fe8.png)

Reply:
fixed 7 lines up